main.go

```diff
@@ -93,46 +96,58 @@ func main() {
 
 	version := "init"
 	lastUpdated := time.Now()
-	for {
-		newVersion, err := m.GetVersion()
-		update := false
 
+	ticker := time.NewTicker(time.Duration(pollInterval) * time.Millisecond)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		update, updateForced := false, false
+		newVersion, err := m.GetVersion()
 		if err != nil {
-			logrus.Errorf("Error reading version: %v", err)
+			logrus.Errorf("Failed to get metadata version: %v", err)
 		} else if version != newVersion {
-			logrus.Debugf("Version has been changed. Old version: %s. New version: %s.", version, newVersion)
+			logrus.Debugf("Metadata version changed. Old: %s New: %s.", version, newVersion)
 			version = newVersion
 			update = true
 		} else {
-			logrus.Debugf("No changes in version: %s", newVersion)
 			if time.Since(lastUpdated).Minutes() >= forceUpdateInterval {
-				logrus.Debugf("Executing force update as version hasn't been changed in: %v minutes", forceUpdateInterval)
-				update = true
+				logrus.Debugf("Executing force update as metadata version hasn't changed in: %d minutes",
+					forceUpdateInterval)
+				updateForced = true
 			}
 		}
 
-		if update {
+		if update || updateForced {
 			// get records from metadata
 			metadataRecs, err := m.GetMetadataDnsRecords()
 			if err != nil {
-				logrus.Errorf("Error reading external dns entries: %v", err)
+				logrus.Errorf("Failed to get DNS records from metadata: %v", err)
 			}
-			logrus.Debugf("DNS records from metadata: %v", metadataRecs)
 
-			//update provider
-			updated, err := UpdateProviderDnsRecords(metadataRecs)
-			if err != nil {
-				logrus.Errorf("Failed to update provider with new DNS records: %v", err)
-			}
+			logrus.Debugf("DNS records from metadata: %v", metadataRecs)
 
-			for _, toUpdate := range updated {
-				serviceDnsRecord := utils.ConvertToServiceDnsRecord(toUpdate)
-				c.UpdateServiceDomainName(serviceDnsRecord)
+			// A flapping service might cause the metadata version to change
+			// in short intervals. Caching the previous metadata DNS records
+			// allows us to check if the actual records have changed, before
+			// querying the provider records.
+			if !reflect.DeepEqual(metadataRecs, metadataRecsCached) || updateForced {
```

Comment:
Can we swap the order, and put updateForce first? So when updateForce=true, DeepEqual operation gets skipped